File: 49-group-anagrams/group-anagrams.java
class Solution {
    public List<List<String>> groupAnagrams(String[] strs) {
        Map<String,List<String>> groupMap = new HashMap<>();
        for(String s:strs){
            char[] ch = s.toCharArray();
            Arrays.sort(ch);
            String key = new String(ch);
            if(!groupMap.containsKey(key)){
                groupMap.put(key,new ArrayList<>());
            }
            groupMap.get(key).add(s);
        }
        return new ArrayList<>(groupMap.values());
    }
}

File: 49-group-anagrams/group-anagrams.py
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        map = {}

        for s in strs:
            sort_str = "".join(sorted(s))
            if sort_str not in map:
                map[sort_str]=[]
            map[sort_str].append(s)

        return list(map.values())